mssqlcli/main.py

```diff
@@ -48,23 +41,36 @@ def run_cli_with(options):
     # Moved import here so we can create the config dir for first use prior.
     from mssqlcli.mssql_cli import MssqlCli
 
-    mssqlcli = MssqlCli(options)
-    mssqlcli.connect_to_database()
+    # set interactive mode to false if -Q is specified
+    if options.query:
+        options.interactive_mode = False
 
-    telemetry_session.set_server_information(mssqlcli.mssqlcliclient_main)
-    mssqlcli.run()
+    mssqlcli = MssqlCli(options)
+    try:
+        mssqlcli.connect_to_database()
+        telemetry_session.set_server_information(mssqlcli.mssqlcliclient_main)
+
+        if mssqlcli.interactive_mode:
+            # run cli if in interactive mode

```

Comment:
we can remove these code comments as the code is already explicit as to whats going on and the comments aren't adding any value.